modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1408,7 +1408,8 @@ else if (!F.isEmpty(req.warnings())) {
             }
 
             return new SnapshotOperationResponse();
-        }, cctx.kernalContext().pools().getSnapshotExecutorService());
+        }, Optional.ofNullable(cctx.kernalContext().pools().getSnapshotExecutorService())
```

Comment:
Looks very strange. In this place snpExecSvc must not be null. Let' check it in `createSnapshot`?